src/main/groovy/life/qbic/portal/sampletracking/ui/SampleImport.groovy

```diff
@@ -12,29 +17,133 @@ class SampleImport extends VerticalLayout {
 
     private Button singleSampleAddButton
     private TextField additionalSampleId
+    private Upload fileSampleAddUpload
 
     SampleImport(PortletController controller, SampleImportModel sampleImportModel) {
         super()
         this.controller = controller
         this.sampleImportModel = sampleImportModel
         initLayout()
+        registerListeners()
     }
 
     private def initLayout() {
 
         FormLayout formLayout = new FormLayout()
+        // Add textfield for sample Id input with placeholder sample id
         this.additionalSampleId = new TextField("Sample ID")
         this.additionalSampleId.setPlaceholder("QABCD004AO")
+        // Add upload receiver to get uploaded file content
+        Uploader receiver = new Uploader()
+        // Add upload field for sample upload via file
+        this.fileSampleAddUpload = new Upload("Upload File here", receiver)
+        //Start upload only after add samples button was pressed
+        this.fileSampleAddUpload.setImmediateMode(false)
+        this.fileSampleAddUpload.setButtonCaption("Add File IDs")
+
+        //TODO: Restrict Fileupload to specific Type and maximum size,
+        // add possibility of progressbar?
+
+        // Add Button to add sample
         this.singleSampleAddButton = new Button("Add Sample")
+        // Add components to layout
+        formLayout.addComponents(this.additionalSampleId, this.singleSampleAddButton, this.fileSampleAddUpload)
 
-        formLayout.addComponents(this.additionalSampleId, this.singleSampleAddButton)
-        // add components
         this.addComponents(formLayout)
     }
 
     private def registerListeners() {
+        // Add listener to add sample button
```

Comment:
```suggestion
```
I think that is obvious from the code and comments such as this only make the code harder to read / clutter the code.